core/src/core/pmmr/segment.rs

```diff
@@ -132,6 +132,7 @@ impl<T> Segment<T> {
 		(first, last)
 	}
 
+	/// TODO - binary_search_by_key() here (can we assume these are sorted by pos?)
```

Comment:
We can't use `binary_search_by_key` because we only have an iterator here, not a slice